shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1277,52 +1284,46 @@ - (void)updateViewportPadding {
 
 #pragma mark - Keyboard events
 
+typedef NS_ENUM(NSInteger, FlutterKeyboardMode) {
+  FlutterKeyboardModeHidden = 0,
+  FlutterKeyboardModeDocked = 1,
+  FlutterKeyboardModeFloating = 2,
+};
+
 - (void)keyboardWillShowNotification:(NSNotification*)notification {
   // Immediately prior to a docked keyboard being shown or when a keyboard goes from
-  // undocked/floating to docked, this notification is triggered
-
-  NSDictionary* info = [notification userInfo];
-  CGRect keyboardFrame = [[info objectForKey:UIKeyboardFrameEndUserInfoKey] CGRectValue];
-  bool isEmpty = CGRectIsEmpty(keyboardFrame);
-
-  // If keyboard is empty, bypass check if it's from another app
-  if (isEmpty == false) {
-    // Ignore keyboard notifications related to other apps.
-    id isLocal = info[UIKeyboardIsLocalUserInfoKey];
-    if (isLocal && ![isLocal boolValue]) {
-      return;
-    }
+  // undocked/floating to docked, this notification is triggered.
+  [self handleKeyboardNotification:notification notificationName:UIKeyboardWillShowNotification];
```

Comment:
Can you not trust `notification.name` for some reason?
```suggestion
  [self handleKeyboardNotification:notification notificationName:notification.name];
```

Reply:
Can probably just remove the `notificationName:` parameter